monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/micro/integrator/delegates/TemplatesDelegate.java

```diff
@@ -20,29 +20,148 @@
+            count = DelegatesUtil.getArtifactCount(searchedList);
+        }
+        Artifacts paginatedList = DelegatesUtil.getPaginationResults(searchedList, fromIndex, toIndex);
+        ArtifactsResourceResponse artifactsResourceResponse = new ArtifactsResourceResponse();
+        artifactsResourceResponse.setResourceList(paginatedList);
+        artifactsResourceResponse.setCount(count);
+        prevSearchKey = searchKey;
+        return artifactsResourceResponse;
+    }
+
+     
+    public static List<ArtifactsInner> getSearchedTemplatesResultsFromMI(String groupId, List<String> nodeList, 
+       String searchKey, String order, String orderBy)
+            throws ManagementApiException {
+        
+        Artifacts artifacts = new Artifacts();
+
+        for (String nodeId: nodeList) {
+
+            String mgtApiUrl = ManagementApiUtils.getMgtApiUrl(groupId, nodeId);
+            String accessToken = DATA_MANAGER.getAccessToken(groupId, nodeId);
+            
+            JsonArray artifactList = DelegatesUtil.getResourceResultList(groupId, nodeId, Constants.TEMPLATES,
+                mgtApiUrl, accessToken, searchKey);
+            
+
+            for (JsonElement jsonElement : artifactList) {
+                JsonObject artifact = (JsonObject) jsonElement;
+                String templateName = artifact.get("name").getAsString();
+                String templateType = artifact.get("type").getAsString();
+                
+                ArtifactDetails artifactDetails = getTemplateDetails(groupId, nodeId, templateType, templateName,
+                                                                     mgtApiUrl, accessToken);
+                                                                     
+                AtomicBoolean isRecordExist = new AtomicBoolean(false);
+                artifacts.stream().filter(o -> (o.getName()).equals(templateName)).forEach(
+                        o -> {
+                            o.getNodes().add(artifactDetails);
+                            isRecordExist.set(true);
+                        });
+                if (!isRecordExist.get()) {
+                    ArtifactsInner artifactsInner = new ArtifactsInner();
+                    artifactsInner.setName(templateName);
+                    List<ArtifactDetails> artifactDetailsList = new ArrayList<>();
+                    artifactDetailsList.add(artifactDetails);
+                    artifactsInner.setNodes(artifactDetailsList);
+                    artifactsInner.setType(templateType);
+                    artifacts.add(artifactsInner);
+                }
+            }
+        }
+
+        //ordering   
+        Comparator<ArtifactsInner> comparatorObject;
+        switch (orderBy.toLowerCase()) {
+            //for any other ordering options
+            default: comparatorObject = Comparator.comparing(ArtifactsInner::getNameIgnoreCase); break;
+        }
+        if (order.equalsIgnoreCase("desc")) {
```

Comment:
If order can be null let's swich the two parameters

Reply:
fixed